clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -1200,13 +1198,23 @@ private void completeOffsetCommitRequestWithError(CommitRequestManager commitReq
     }
 
     private void testRetriable(final CommitRequestManager commitRequestManager,
-                               final List<CompletableFuture<Map<TopicPartition, OffsetAndMetadata>>> futures) {
+                               final List<CompletableFuture<Map<TopicPartition, OffsetAndMetadata>>> futures,
+                               final Errors error
+    ) {
         futures.forEach(f -> assertFalse(f.isDone()));
 
-        // The manager should backoff for 100ms
-        time.sleep(100);
-        commitRequestManager.poll(time.milliseconds());
+        // The manager should backoff before retry
+        time.sleep(retryBackoffMs);
+        NetworkClientDelegate.PollResult poll = commitRequestManager.poll(time.milliseconds());
+        assertEquals(1, poll.unsentRequests.size());
         futures.forEach(f -> assertFalse(f.isDone()));
+        poll.unsentRequests.get(0).handler().onComplete(buildOffsetFetchClientResponse(poll.unsentRequests.get(0), new HashSet<>(), error));
+
+        // Sleep util timeout
+        time.sleep(defaultApiTimeoutMs);
+        poll = commitRequestManager.poll(time.milliseconds());
+        assertEquals(1, poll.unsentRequests.size());
+        futures.forEach(f -> assertTrue(f.isCompletedExceptionally()));
```

Comment:
Could you please check the cause of exception? that should be `TimeoutException`

Reply:
Thanks for pointing out, I will utilize `assertFutureThrows`  to achieve that. :smile: